lua/starfall/libs_sh/darkrp.lua

```diff
@@ -0,0 +1,1119 @@
+local checkluatype = SF.CheckLuaType
+local checkpattern = SF.CheckPattern
+local registerprivilege = SF.Permissions.registerPrivilege
+
+-- Under normal circumstances, an API change could introduce security
+-- vulnerabilities. Suppose a DarkRP function changes to also return a Player
+-- or Entity object, whereas before it only returned safe types like numbers
+-- and strings. Starfall code unaware of the change might expose that function
+-- in a way that ends in a tail call, which would result in all values being
+-- returned as-is. That would be bad, as it would allow a guest to obtain
+-- unwrapped objects.
+-- checksafety and assertsafety resolve this by ensuring that unsafe values
+-- don't make it to the guest. The way they are implemented have the downside
+-- of those values not making it to the client at all, when they could be
+-- wrapped. I could have done something like this to mitigate that:
+--return unwrap(instance.Sanitize({...}))
+-- ...but since nearly every single hook and function (in this file) is going
+-- to be using these functions, I wanted to keep the added overhead as small
+-- as possible.
+local checksafety, assertsafety
+do
+	local whitelist = {
+		["nil"] = true,
+		["boolean"] = true,
+		["number"] = true,
+		["string"] = true,
+	}
+	local pairs = pairs
+	local type = type
+	function checksafety(...)
+		for k, v in pairs({...}) do -- 'pairs' was determined to be faster than 'select'
+			if not whitelist[type(v)] then
+				return false
+			end
+		end
+		return true
+	end
+	local table_remove = table.remove
+	local unpack = unpack
+	function assertsafety(...)
+		local args = {...}
+		for i=#args, 1, -1 do
+			if not whitelist[type(args[i])] then
+				table_remove(args, i)
+			end
+		end
+		return unpack(args)
+	end
+end
+
+if SERVER then
+	timer.Simple(0, function()
+		if util.NetworkStringToID("sf_moneyrequest") ~= 0 then
```

Comment:
loganlearner's DarkRP library creates that network string in a file with the same path as ours. If loganlearner's library takes over and overrides this file, this code will never get run, and if it doesn't, then this condition will not be true. That's why I had it in `sflib.lua`.

Reply:
Does it matter which lib exists? They're likely the same api right? Users who want the more up to date one can request the old one be uninstalled. I suppose this code should be removed entirely then.